app/models/ownership.rb

```diff
@@ -72,4 +72,17 @@ def unconfirmed?
   def safe_destroy
     destroy if unconfirmed? || rubygem.owners.many?
   end
+
+  def validate_unique_user
+    return unless rubygem && user
+    ownerships = persisted? ? Ownership.where.not(id: id) : Ownership
+    other = ownerships.find_by(rubygem:, user:)
+    return unless other
+
+    if other.confirmed?
+      errors.add :user_id, "is already an owner of this gem"
```

Comment:
please put these strings into `en.yml` so they can be translated. (also potentially include `other.display_id` in the error message?)

Reply:
Got it. Thanks for reminding me.